pkgs/applications/graphics/mandelbulber2/default.nix

```
@@ -0,0 +1,38 @@
+{ lib, mkDerivation, fetchFromGitHub, libpng, gsl, libsndfile, lzo,
+qmake, qttools, qtbase, qtmultimedia,
+opencl ? true, opencl-clhpp ? null, ocl-icd ? null }:
+
+assert opencl -> opencl-clhpp != null;
+assert opencl -> ocl-icd != null;
+
+mkDerivation rec {
+  pname = "mandelbulber2";
+  version = "2.20";
+
+  src = fetchFromGitHub {
+    owner = "buddhi1980";
+    repo = "mandelbulber2";
+    rev = version;
+    sha256 = "043dks9fimhradyhdzqdc6lb9z0x9lkj3szj10751g424lppp207";
+  };
+
+  nativeBuildInputs = [ qmake qttools ];
+  buildInputs = [ qtbase qtmultimedia libpng gsl libsndfile lzo ] ++
+    lib.optionals opencl [ opencl-clhpp ocl-icd ];
+
+  sourceRoot = "source/mandelbulber2";
```

```suggestion
  sourceRoot = "${src.name}/mandelbulber2";
```

might be more reliable (e.g. when one overrides `src` to point to a local directory for testing a patch)